conjure-java-core/src/main/java/com/palantir/conjure/java/types/UnionGenerator.java

```diff
@@ -542,7 +542,7 @@ private static TypeSpec generateBase(
                                     .addAnnotation(AnnotationSpec.builder(JsonCreator.class)
                                             .build())
                                     .addParameter(ParameterSpec.builder(memberType, VALUE_FIELD_NAME)
-                                            .addAnnotation(jsonPropertyAnnotation)
+                                            .addAnnotation(wrapperConstructorParameterAnnotation(memberTypeDef))
```

Comment:
does this JsonSetter approach also work for aliases? like if someone made a union variant which is an Alias of `List<String>` e.g. a Poem where each string is a line?

```yaml
Performance:
  union
    poem: Poem
    dance: Dance
```

Reply:
Unfortunately not, that gets a little more complicated. The difficulty is hinting that we want an alias of null rather than a null alias.
In the optional case we generate a default no-arg constructor, we would need to do something similar for the collection types.